exonum-java-binding/qa-service/src/test/java/com/exonum/binding/qaservice/QaServiceImplIntegrationTest.java

```diff
@@ -114,356 +88,394 @@ void tearDown() {
-    });
+    short validatorCount = 8;
+    try (TestKit testKit = TestKit.builder(EmulatedNodeType.VALIDATOR)
+        .withValidators(validatorCount)
+        .withService(QaServiceModule.class)
+        .build()) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      StoredConfiguration configuration = service.getActualConfiguration();
+      EmulatedNode emulatedNode = testKit.getEmulatedNode();
+      PublicKey emulatedNodeServicePublicKey = emulatedNode.getServiceKeyPair().getPublicKey();
+      List<ValidatorKey> validatorKeys = configuration.validatorKeys();
+
+      assertThat(validatorKeys).hasSize(validatorCount);
+      assertThat(emulatedNodeServicePublicKey).matches(pk -> validatorKeys.stream()
+          .anyMatch(v -> v.serviceKey().equals(pk)));
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getBlockTransactionsByBlockId() {
-    withNodeFake(() -> {
-      Throwable t = assertThrows(RuntimeException.class, () -> service.getBlockTransactions(0L));
-      assertThat(t.getMessage()).contains(NO_GENESIS_BLOCK_ERROR_MESSAGE);
-    });
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
+      List<HashCode> blockTransactions = service.getBlockTransactions(1L);
+      assertThat(blockTransactions).hasSize(1);
+      assertThat(blockTransactions.get(0)).isEqualTo(createCounterTransaction.hash());
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxMessages() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionMessage> txMessages = service.getTxMessages();
-      assertThat(txMessages).isEmpty();
-    });
+      assertThat(txMessages).hasSize(2);
+      assertThat(txMessages.get(createCounterTransaction.hash()))
+          .isEqualTo(createCounterTransaction);
+    }
   }
 
   @Test
-  @RequiresNativeLibrary
   void getTxResults() {
-    withNodeFake(() -> {
+    try (TestKit testKit = TestKit.forService(QaServiceModule.class)) {
+      QaServiceImpl service = testKit.getService(QaService.ID, QaServiceImpl.class);
+      TransactionMessage createCounterTransaction = createCreateCounterTransaction("counterName");
+      testKit.createBlockWithTransactions(createCounterTransaction);
       Map<HashCode, TransactionResult> txResults = service.getTxResults();
-      assertThat(txResults).isEmpty();
-    });
+      assertThat(txResults).hasSize(1);
```

Comment:
`….isEqualTo(Map.of(createCounterTransaction.hash(),  TransactionResult.successful());`